src/codegen/codegen_opencl.cc

```diff
@@ -208,6 +208,16 @@ void CodeGenOpenCL::VisitExpr_(const Broadcast* op, std::ostream& os) {   // NOL
   os << "))";
 }
 
+void CodeGenOpenCL::VisitExpr_(const Call *op, std::ostream& os) {  // NOLINT(*)
+  if (op->is_intrinsic(intrinsic::tvm_if_then_else)) {
```

Comment:
Please add a comment on why explicit cast is needed here.

Likely you also need to add support for select

Reply:
Please add a regression test case as per https://docs.tvm.ai/contribute/code_review.html#ensure-test-coverage

Reply:
Tests added for both tvm::if_then_else and Select